### PR TITLE
feat(content): Adds mushroom farming via racks and logs, obsoletes spores

### PR DESCRIPTION
**Pull request by Ali-Anomma**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

### Optional

- [ ] This PR ports commits from DDA or other cataclysm forks.
  - [ ] I have attributed original authors in the commit messages adding [`Co-Authored-By`](https://docs.github.com/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) in the commit message.
  - [ ] I have linked the URL of original PR(s) in the description.
- [ ] This is a C++ PR that modifies JSON loading or behavior.
  - [ ] I have documented the changes in the appropriate location in the `doc/` folder.
  - [ ] If documentation for this feature does not exist, please write it or at least note its lack in PR description.
  - [ ] New localizable fields need to be added to the `lang/bn_extract_json_strings.sh` script if it does not support them yet.
  - [ ] If applicable, add checks on game load that would validate the loaded data.
  - [ ] If it modifies format of save files, please add migration from the old format.
- [ ] This is a PR that modifies build process or code organization.
  - [ ] Please document the changes in the appropriate location in the `doc/` folder.
  - [ ] If documentation for this feature or process does not exist, please write it.
  - [ ] If the change alters versions of software required to build or work with the game, please document it.

- [X] This is a PR that removes JSON entities.
  - [X] The removed JSON entities have new entries in `data/json/obsoletion/` folder or use some other migration process for existing saves.

## Purpose of change

A macronutritient system idea is floating around for a long time. But if it implemented today, there would be a big lack of protein sources. Mushrooms are a reliable protein source, but their way of farming is just like normal plants, which was a limitation of old code, where "PLANTABLE" flag wasnt jsonized yet.
## Describe the solution

Added mushroom culture as a replacement for seed-like mushroom spore, added racks for artifical mushroom growing medium, while repurposed stumps and trunks for natural alternatives.
## Describe alternatives you've considered

Keep the mushroom spores, or make mushroom culture plantable on dirt
## Testing

Used a local mod for testing
## Additional context
We need a reminder to add protein to mushrooms when we finally add macronutritients

### CODE REVIEW

File: data/json/obsoletion/recipes.json [hunk 1]
```diff
@@ -60,6 +60,21 @@
...
+  {
+    "type": "recipe",
+    "result": "seed_mushroom",
+    "obsolete": true
+  },
+  {
+    "type": "recipe",
+    "result": "cooked_burrito",
+    "obsolete": true
+  },
```
Comment (ChucklesTheBeard) [inline data/json/obsoletion/recipes.json:77]: ?

**Review by RoyalFox2140 — CHANGES_REQUESTED** [review]
You are obsoleting "seed_mushroom" twice on the recipes.json obsoletion file, without obsoleting the recipe for morel spores. You are also either obsoleting "cooked_burrito' or otherwise aren't adding your obsoletions at the end of the file which confused some (including me).

Everything else is fine.

File: data/json/obsoletion/recipes.json (hunk 1 above)
Comment (chaosvolt) [inline data/json/obsoletion/recipes.json:77]:
```suggestion
  {
    "type": "recipe",
    "result": "seed_mushroom",
    "obsolete": true
  },
  {
    "type": "recipe",
    "result": "seed_mushroom_morel",
    "obsolete": true
  },
```

File: data/json/items/comestibles/seed.json [hunk 2]
```diff
@@ -800,5 +776,59 @@
...
+    "charges": 1,
+    "stack_size": 100,
+    "material": "veggy",
+    "primary_material": "mushroom",
+    "symbol": ".",
+    "color": "light_gray",
+    "flags": [ "NUTRIENT_OVERRIDE", "PLANTABLE_SEED", "UNSAFE_CONSUME" ],
+    "id": "mushroom_culture",
+    "copy-from": "seed",
+    "description": "A wet plant chunk, suitable for growing your own mushrooms. You need to \"plant\" it on logs or racks to grow though.",
```
Comment (chaosvolt) [inline data/json/items/comestibles/seed.json:798]:
```suggestion
    "description": "A wet plant chunk, suitable for growing your own mushrooms.  You need to \"plant\" it on logs or racks to grow though.",
```

File: data/json/furniture_and_terrain/terrain-manufactured.json [hunk 3]
```diff
@@ -1495,5 +1495,27 @@
       "items": [ { "item": "pointy_stick", "count": 4 }, { "item": "string_6", "count": 4 }, { "item": "tarp", "count": 1 } ]
     },
     "flags": [ "TRANSPARENT", "FLAMMABLE", "THIN_OBSTACLE", "INDOORS", "MOUNTABLE", "EASY_DECONSTRUCT" ]
+  },
+  {
+    "id": "t_rack_mushroom",
+    "name": "mushroom rack",
+    "description": "A rack specifically made for raising mushrooms. You can plant mushroom cultures in here.",
```
Comment (chaosvolt) [inline data/json/furniture_and_terrain/terrain-manufactured.json:1502]:
```suggestion
    "description": "A rack specifically made for raising mushrooms.  You can plant mushroom cultures in here.",
```

File: data/json/items/comestibles/seed.json [hunk 4]
```diff
@@ -800,5 +776,59 @@
...
+    "charges": 1,
+    "stack_size": 100,
+    "material": "veggy",
+    "primary_material": "mushroom",
+    "symbol": ".",
+    "color": "light_gray",
+    "flags": [ "NUTRIENT_OVERRIDE", "PLANTABLE_SEED", "UNSAFE_CONSUME" ],
+    "id": "mushroom_morel_culture",
+    "copy-from": "seed",
+    "description": "A wet plant chunk, suitable for growing your own morel mushrooms. You need to \"plant\" it on logs or racks to grow.",
```
Comment (chaosvolt) [inline data/json/items/comestibles/seed.json:826]:
```suggestion
    "description": "A wet plant chunk, suitable for growing your own morel mushrooms.  You need to \"plant\" it on logs or racks to grow.",
```

**Review by chaosvolt — COMMENTED** [review]
This is only technically a review just so I can bundle the double-space fixes in one quick commit with less spam.

File: data/json/items/comestibles/seed.json [hunk 5]
```diff
@@ -800,5 +776,59 @@
...
+    "comestible_type": "FOOD",
+    "name": { "str_sp": "morel mushroom culture" },
+    "category": "seeds",
+    "volume": "250 ml",
+    "weight": "100 g",
+    "price": "250 cent",
+    "price_postapoc": "40 cent",
+    "charges": 1,
+    "stack_size": 100,
+    "material": "veggy",
```
Comment (chaosvolt) [inline data/json/items/comestibles/seed.json:819]:
```suggestion
    "name": { "str_sp": "mushroom culture" },
    "weight": "100 g",
    "price": "2 USD",
    "charges": 1,
    "primary_material": "mushroom",
    "color": "light_gray",
    "id": "mushroom_culture",
    "copy-from": "seed",
    "description": "A wet plant chunk, suitable for growing your own mushrooms.  You need to \"plant\" it on logs or racks to grow though.",
    "seed_data": {
      "plant_name": "mushroom",
      "fruit": "mushroom",
      "fruit_div": 2,
      "seeds": false,
      "required_terrain_flag": "MUSHROOM_PLANTABLE",
      "grow": "28 days"
    }
  },
  {
    "type": "COMESTIBLE",
    "name": { "str_sp": "morel mushroom culture" },
    "weight": "100 g",
    "price": "250 cent",
    "price_postapoc": "40 cent",
    "charges": 1,
    "material": "veggy",
```

**Review by chaosvolt — CHANGES_REQUESTED** [review]
Okay so github is screaming at me so I can't just check this PR out and fix it directly, but: having the mushroom rack be terrain is actually really bad. Planting on terrain converts it to dirt, so you'll be wasting your nails and canvas bag every time, plus this can be constructed on any empty floor as a way to convert it to dirt afterward.

So the obvious answer is make it furntiure instead, but at that point question: do you still want specifically mushroom racks to be a thing separate from planter boxes? If yes then it's mostly gonna be a lot of copy-paste from planter boxes and their stages.

If not, tacking the new flag on empty planter boxes would be much simpler solution for that. Only real question becomes whether you want regular seeds and mushrooms to be plantable on the same furniture or not.

**Comment by chaosvolt** [comment]
So it turns out it does work fine but only because right now they can only be planted using zone farming, which is something we're going to want to fix in the future. I guess them being terran is fine for now but I'll likely fuck with replacing them with a furniture version in a followup PR.

**Review by chaosvolt — DISMISSED** [review]

**Review by chaosvolt — APPROVED** [review]
image

**Review by chaosvolt — APPROVED** [review]

**Comment by chaosvolt** [comment]
Figuring out how to unfuck planting via iexamine is my problem now.

**Pull request merged**